cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -542,12 +545,22 @@ func (m *ociManagerImpl) buildNodeFromTemplate(nodePool *oke.NodePool) (*apiv1.N
 		})
 	}
 
+	freeformTags, err := m.ociTagsGetter.GetFreeformTags(nodePool)
+	if err != nil {
+		return nil, err
+	}
 	node.Status.Capacity[apiv1.ResourcePods] = *resource.NewQuantity(110, resource.DecimalSI)
-
 	node.Status.Capacity[apiv1.ResourceCPU] = *resource.NewQuantity(int64(shape.CPU), resource.DecimalSI)
 	node.Status.Capacity[apiv1.ResourceMemory] = *resource.NewQuantity(int64(shape.MemoryInBytes), resource.DecimalSI)
 	node.Status.Capacity[ipconsts.ResourceGPU] = *resource.NewQuantity(int64(shape.GPU), resource.DecimalSI)
 
+	ephemeralStorage, err := getEphemeralResourceRequestsInBytes(freeformTags)
+	if err != nil {
+		klog.V(4).Info(err)
```

Comment:
Hm, curious, why do we log the error, but not return the error?

Reply:
My thinking was that, since it is not mandatory for users to add the required freeform tag, so we could just leave it as an info log with higher verbosity level so it does not spam the logs. This would only be needed if customers want CA to spin up nodes with a specific ephemeral-storage size (especially when they want to scale up the pool from 0).

Let me know if you would still like me to make modifications here.

Reply:
I think if a user is specifying a value, but it is not getting parsed correctly, I personally think we should error out so that the user can quickly know that something is wrong and correct it.

Otherwise, they will think everything is successful, even when it's not.

However, I could see arguments where it would be okay to warn the user. For example, the autoscaler is working, then the user mistypes a free form tag which will cause a failure, but doesn't see a failure until days later when a scale operation actually occurs. Or the fact that specifying one wrong value will cause scaling to not happen at all.

Overall, I'd be okay with erroring out or warning (but slightly prefer erroring).

Reply:
Agreed, thanks. I've changed the logic here.